### PR TITLE
Support for process output on detail page

### PR DESCRIPTION
**Pull request by MarieVerdonck**
## References
* Depends on: https://github.com/DSpace/DSpace/pull/2934

## Description
This PR shows the output of a Process on the process' detail page at {dspaceUI}/processes/<process-id>

## Instructions for Reviewers
The logs of the process are are a list of strings, shown underneath each other is a < pre > section.
If the process has no ProcessOutput or the logs in the ProcessOutput is just an empty array, then a message is shown.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs for any bug fixes, improvements or new features. A few reminders about what constitutes good tests:
    * Include tests for different user types (if behavior differs), including: (1) Anonymous user, (2) Logged in user (non-admin), and (3) Administrator.
    * Include tests for error scenarios, e.g. when errors/warnings should appear (or buttons should be disabled).
    * For bug fixes, include a test that reproduces the bug and proves it is fixed. For clarity, it may be useful to provide the test in a separate commit from the bug fix.
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/app/core/data/process-output-data.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    protected objectCache: ObjectCacheService,
+    protected halService: HALEndpointService,
+    protected notificationsService: NotificationsService,
+    protected http: HttpClient,
+    protected comparator: DefaultChangeAnalyzer<ProcessOutput>) {
+    super();
+  }
+}
+
+// @ts-ignore
```
[the remark this answers is not in the document]
Comment (MarieVerdonck) [reply src/app/core/data/process-output-data.service.ts]: Got in there by accident, removed.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@MarieVerdonck : I re-tested this with the backend PR (https://github.com/DSpace/DSpace/pull/2934).  It's not quite working as expected, though it's no longer throwing obvious errors. 

Here's the behavior I see:
* First, I kick off a new process from http://localhost:4000/processes/
* I immediately click on the new process while the process is in the "Running" status.
     * **Bug**: While the process is "running", I see the Process Output section &  "Retrieve process output" button...even though no output is yet available.
* **Bug:** Once the process is "Completed", I click on the "Retrieve process output" and get a response that "no process output is available". But, if I look at the Chrome DevTools, I see that the `200` response included a "content" link which allows me to download the content directly from the REST API. So, it looks like the UI is not finding that "content" link.
* **Bug:** If I go to an old process (one with no output), and click on "Retrieve process output", I just see the "Retrieving..." message forever.  If I look at Chrome DevTools, I see it responded with a `204 No Content` response, but it looks like the UI didn't handle that response.

**Comment by tdonohue** [comment]
@MarieVerdonck : Could you rebase this PR on the latest code?  It looks like there's a small merge conflict here now & I'd like to be able to re-test this PR with the new backend PR (https://github.com/DSpace/DSpace/pull/3009)

**Review by tdonohue — APPROVED** [review]
👍 @MarieVerdonck : I've re-reviewed & retested this today with the new backend PR (https://github.com/DSpace/DSpace/pull/3009) and it works exactly as described (all my previously reported issues are gone as well).  The code looks good too.  Thanks!

**Pull request merged**